### PR TITLE
core: pathfinding: log a little extra info about failures

### PR DESCRIPTION
**Pull request by eckter**
We barely logged anything on pathfindings. We now log:

1. success
2. no path found at all, including the last reached waypoint index
3. no path found because of [constraint type / error type]

Not much, but it can still help

This PR also fixes the "couldn't find route" error, it had an irrelevant type.

### CODE REVIEW

File: core/src/main/kotlin/fr/sncf/osrd/graph/Pathfinding.kt [hunk 1]
```diff
@@ -184,15 +186,23 @@ class Pathfinding<NodeT : Any, EdgeT : Any, OffsetType>(
...
         }
+        var maxReachedTarget = 0
         val start = Instant.now()
         while (true) {
             if (Duration.between(start, Instant.now()).toSeconds() >= timeout)
                 throw OSRDError(ErrorType.PathfindingTimeoutError)
-            val step = queue.poll() ?: return null
+            val step = queue.poll()
+            if (step == null) {
+                pathfindingLogger.info(
```
Comment (bougue-pe) [inline core/src/main/kotlin/fr/sncf/osrd/graph/Pathfinding.kt:196]: Should we downgrade log level to debug for this (and maybe some others on happy path)?
I wonder if it'll spam logs (maybe up the aws costs a bit, but mostly make logs less readable).

**Review by bougue-pe — APPROVED** [review]
LGTM (feel free to resolve comment)

**Review by Erashin — APPROVED** [review]

File: core/src/main/kotlin/fr/sncf/osrd/graph/Pathfinding.kt (hunk 1 above)
Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/graph/Pathfinding.kt:196]: The logs in this PR can only display something once per request, which doesn't feel excessive. (per line, maybe 3 lines total per request max over the PR)

If we want to be more careful about what we log, maybe we should start with the warnings when loading an infra... Actually I'll open an issue for this

Comment (eckter) [reply core/src/main/kotlin/fr/sncf/osrd/graph/Pathfinding.kt:196]: https://github.com/OpenRailAssociation/osrd/issues/8773

**Pull request merged**